challenge_8/scala/ndsmith3/WeirdLinkedList.scala

```diff
@@ -0,0 +1,87 @@
+import scala.util.Random.nextInt
+
+case class WeirdList[A](data: A) {
+  var next: Option[WeirdList[A]] = None
+  var rand: Option[WeirdList[A]] = None
+  var length: Int                = 1
+
+  def add(data: A): Unit = {
+    def assignRand(list: WeirdList[A])(
+        index: Int = nextInt(this.length)): Unit = {
+      if (index == this.indexOf(list.data)) assignRand(list)()
+      else list.rand = Option(this.subList(index))
+    }
+
+    if (this.next.isDefined) this.next.get.add(data)
+    else this.next = Option(WeirdList(data))
+    if (this.length > 1) this foreach (x => assignRand(x)())
+    length += 1
+  }
+
+  def addNode(node: WeirdList[A]): Unit = {
+    if (this.next.isDefined) this.next.get.addNode(node)
+    else this.next = Option(node)
+  }
```

Comment:
> if (this.next.isDefined) this.next.get.addNode(node)
> else this.next = Option(node)

isDefined and get are code smells.  I would see if this works:

this.next.foreach(_.addNode(node)).orElse(this.next = Option(node))